test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Fetching/FetchResourceTests.cs

```diff
@@ -0,0 +1,150 @@
+
+            var item2 = responseDocument.ManyData.Single(resource => resource.Id == workItems[1].StringId);
+            item2.Type.Should().Be("workItems");
+            item2.Attributes["description"].Should().Be(workItems[1].Description);
+            item2.Attributes["dueAt"].Should().BeCloseTo(workItems[1].DueAt);
+            item2.Attributes["priority"].Should().Be(workItems[1].Priority.ToString("G"));
+        }
+
+        [Fact]
+        public async Task Can_get_primary_resource_by_ID()
+        {
+            // Arrange
+            var workItem = _fakers.WorkItem.Generate();
+
+            await _testContext.RunOnDatabaseAsync(async db =>
+            {
+                await db.GetCollection<WorkItem>().InsertOneAsync(workItem);
+            });
+
+            var route = "/workItems/" + workItem.StringId;
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<Document>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            responseDocument.SingleData.Should().NotBeNull();
+            responseDocument.SingleData.Type.Should().Be("workItems");
+            responseDocument.SingleData.Id.Should().Be(workItem.StringId);
+            responseDocument.SingleData.Attributes["description"].Should().Be(workItem.Description);
+            responseDocument.SingleData.Attributes["dueAt"].Should().BeCloseTo(workItem.DueAt);
+            responseDocument.SingleData.Attributes["priority"].Should().Be(workItem.Priority.ToString("G"));
+        }
+
+        [Fact]
+        public async Task Cannot_get_primary_resource_for_unknown_type()
+        {
+            // Arrange
+            var route = "/doesNotExist/5f88857c4aa60defec6a4999";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<string>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.NotFound);
+
+            responseDocument.Should().BeEmpty();
+        }
+
+        [Fact]
+        public async Task Cannot_get_primary_resource_for_unknown_ID()
+        {
+            // Arrange
+            var route = "/workItems/5f88857c4aa60defec6a4999";
+
+            // Act
+            var (httpResponse, responseDocument) = await _testContext.ExecuteGetAsync<ErrorDocument>(route);
+
+            // Assert
+            httpResponse.Should().HaveStatusCode(HttpStatusCode.NotFound);
+
+            responseDocument.Errors.Should().HaveCount(1);
+            responseDocument.Errors[0].StatusCode.Should().Be(HttpStatusCode.NotFound);
+            responseDocument.Errors[0].Title.Should().Be("The requested resource does not exist.");
+            responseDocument.Errors[0].Detail.Should().Be("Resource of type 'workItems' with ID '5f88857c4aa60defec6a4999' does not exist.");
+        }
+        
+        [Fact]
+        public async Task Cannot_get_secondary_HasOne_resource()
```

Comment:
We should have a test for HasMany too, for example: GET /workItems/123/subscribers.